pkg/clusterapi/aws/client.go

```diff
@@ -0,0 +1,115 @@
+	"github.com/aws/aws-sdk-go/service/elb"
+	"github.com/aws/aws-sdk-go/service/elb/elbiface"
+
+	cov1 "github.com/openshift/cluster-operator/pkg/apis/clusteroperator/v1alpha1"
+)
+
+//go:generate mockgen -source=./client.go -destination=./client_generated_test.go -package=aws
+
+// Client is a wrapper object for actual AWS SDK clients to allow for easier testing.
+type Client interface {
+	DescribeImages(*ec2.DescribeImagesInput) (*ec2.DescribeImagesOutput, error)
+	DescribeVpcs(*ec2.DescribeVpcsInput) (*ec2.DescribeVpcsOutput, error)
+	DescribeSubnets(*ec2.DescribeSubnetsInput) (*ec2.DescribeSubnetsOutput, error)
+	DescribeSecurityGroups(*ec2.DescribeSecurityGroupsInput) (*ec2.DescribeSecurityGroupsOutput, error)
+	RunInstances(*ec2.RunInstancesInput) (*ec2.Reservation, error)
+	DescribeInstances(*ec2.DescribeInstancesInput) (*ec2.DescribeInstancesOutput, error)
+	TerminateInstances(*ec2.TerminateInstancesInput) (*ec2.TerminateInstancesOutput, error)
+
+	RegisterInstancesWithLoadBalancer(*elb.RegisterInstancesWithLoadBalancerInput) (*elb.RegisterInstancesWithLoadBalancerOutput, error)
+}
+
+type awsClient struct {
+	ec2Client ec2iface.EC2API
+	elbClient elbiface.ELBAPI
+}
+
+func (c awsClient) DescribeImages(input *ec2.DescribeImagesInput) (*ec2.DescribeImagesOutput, error) {
+	return c.ec2Client.DescribeImages(input)
+}
+
+func (c awsClient) DescribeVpcs(input *ec2.DescribeVpcsInput) (*ec2.DescribeVpcsOutput, error) {
+	return c.ec2Client.DescribeVpcs(input)
+}
+
+func (c awsClient) DescribeSubnets(input *ec2.DescribeSubnetsInput) (*ec2.DescribeSubnetsOutput, error) {
+	return c.ec2Client.DescribeSubnets(input)
+}
+
+func (c awsClient) DescribeSecurityGroups(input *ec2.DescribeSecurityGroupsInput) (*ec2.DescribeSecurityGroupsOutput, error) {
+	return c.ec2Client.DescribeSecurityGroups(input)
+}
+
+func (c awsClient) RunInstances(input *ec2.RunInstancesInput) (*ec2.Reservation, error) {
+	return c.ec2Client.RunInstances(input)
+}
+
+func (c awsClient) DescribeInstances(input *ec2.DescribeInstancesInput) (*ec2.DescribeInstancesOutput, error) {
+	return c.ec2Client.DescribeInstances(input)
+}
+
+func (c awsClient) TerminateInstances(input *ec2.TerminateInstancesInput) (*ec2.TerminateInstancesOutput, error) {
+	return c.ec2Client.TerminateInstances(input)
+}
+
+func (c awsClient) RegisterInstancesWithLoadBalancer(input *elb.RegisterInstancesWithLoadBalancerInput) (*elb.RegisterInstancesWithLoadBalancerOutput, error) {
+	return c.elbClient.RegisterInstancesWithLoadBalancer(input)
+}
+
+// NewAWSClient creates our client wrapper object for the actual AWS clients we use.
+// For authentication the underlying clients will use either the cluster AWS credentials
+// secret if defined (i.e. in the root cluster),
+// otherwise the IAM profile of the master where the actuator will run. (target clusters)
+func NewAWSClient(kubeClient kubernetes.Interface, mSpec *cov1.MachineSetSpec, namespace, region string) (Client, error) {
```

Comment:
s/NewAWSClient/NewClient/, unless you are going to use awsClient as the return type.